### ISSUE TITLE
Helm upgrade to 1.6.0 fails: cannot patch "aad-pod-identity-nmi" with kind DaemonSet

### ISSUE THREAD

**Issue by robinmanuelthiel** [Azure/aad-pod-identity#600]
**Describe the bug**

When upgrading the Helm chart from version `1.5.5` to `1.6.0`, the following error occurs in three different clusters for me:

```bash
$ helm upgrade aad-pod-identity aad-pod-identity/aad-pod-identity \                                                                                                                               
        --install --wait \
        --namespace aad-pod-identity \
        --version 1.6.0

manifest_sorter.go:175: info: skipping unknown hook: "crd-install"
manifest_sorter.go:175: info: skipping unknown hook: "crd-install"
manifest_sorter.go:175: info: skipping unknown hook: "crd-install"
manifest_sorter.go:175: info: skipping unknown hook: "crd-install"
Error: UPGRADE FAILED: cannot patch "aad-pod-identity-nmi" with kind DaemonSet: DaemonSet.apps "aad-pod-identity-nmi" is invalid: spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"app.kubernetes.io/component":"nmi", "app.kubernetes.io/instance":"aad-pod-identity", "app.kubernetes.io/name":"aad-pod-identity"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable && cannot patch "aad-pod-identity-mic" with kind Deployment: Deployment.apps "aad-pod-identity-mic" is invalid: spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"app.kubernetes.io/component":"mic", "app.kubernetes.io/instance":"aad-pod-identity", "app.kubernetes.io/name":"aad-pod-identity"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable
```

**Steps To Reproduce**

1. Install Helm Chart version 1.5.5
   ```bash
   kubectl create ns aad-pod-identity

   helm upgrade aad-pod-identity aad-pod-identity/aad-pod-identity \
        --install --wait \
        --namespace aad-pod-identity \
        --version 1.5.5
   ```

2. Upgrade to Helm Chart version 1.6.0
   ```bash
   helm upgrade aad-pod-identity aad-pod-identity/aad-pod-identity \
        --install --wait \
        --namespace aad-pod-identity \
        --version 1.6.0
   ```

**Expected behavior**
Upgrade works

**AAD Pod Identity version**
1.6.0

**Kubernetes version**
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.2", GitCommit:"52c56ce7a8272c798dbc29846288d7cd9fbae032", GitTreeState:"clean", BuildDate:"2020-04-16T23:35:15Z", GoVersion:"go1.14.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.10", GitCommit:"059c666b8d0cce7219d2958e6ecc3198072de9bc", GitTreeState:"clean", BuildDate:"2020-04-03T15:17:29Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"linux/amd64"}

**Additional context**

**Comment by robinmanuelthiel** [comment]
My current workaround was uninstalling the old Helm chart and installing the new version, which works but is quite nasty from a CI/CD perspective.

**Comment by aramase** [comment]
`1.5.5` helm chart had introduced 2 labels which could possibly change with chart upgrade - 

```yaml
      app.kubernetes.io/managed-by: Helm
      helm.sh/chart: aad-pod-identity-1.5.5
```

This has been fixed in chart version 1.5.6 to prevent any issues with future upgrades of helm chart. For upgrading from 1.5.5 to any new chart version, a suggested workaround is editing the nmi and mic manifests to remove those 2 from the selector.matchLabels

```
➜ kubectl get ds  aad-pod-identity-nmi -o jsonpath='{.spec.selector.matchLabels}'
map[app.kubernetes.io/component:nmi app.kubernetes.io/instance:pod-identity app.kubernetes.io/managed-by:Helm app.kubernetes.io/name:aad-pod-identity helm.sh/chart:aad-pod-identity-1.5.5]

➜ kubectl edit ds aad-pod-identity-nmi
(Remove `app.kubernetes.io/managed-by: Helm` and `helm.sh/chart: aad-pod-identity-1.5.5` from the spec.selector.matchLabels)
```

The same thing needs to be done with mic too -

```
➜ kubectl get deploy aad-pod-identity-mic -o jsonpath='{.spec.selector.matchLabels}'
map[app.kubernetes.io/component:mic app.kubernetes.io/instance:pod-identity app.kubernetes.io/managed-by:Helm app.kubernetes.io/name:aad-pod-identity helm.sh/chart:aad-pod-identity-1.5.5]

➜ kubectl edit deploy aad-pod-identity-mic
(Remove `app.kubernetes.io/managed-by: Helm` and `helm.sh/chart: aad-pod-identity-1.5.5` from the spec.selector.matchLabels)
```

Once this is done, the helm upgrade command should succeed. I'll also document this workaround in the readme.

**Comment by robinmanuelthiel** [comment]
Thanks for the reply. It would be helpful to have an "Upgrade Guide" or something similar in the docs, as you supposed! 👍
